### PR TITLE
Made tests for sources handlers and withInderexisthandler

### PR DESCRIPTION
**Pull request by DanyilTerentiev**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @Kasterov  @ormykhalyshyn @VitaliiKulinskyi 

### Second Level Review
#
- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by Tysyatsky — APPROVED** [review]
Unit test at their finest. Great work :)

File: Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetCategoriesByStreetcodeIdTest.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using Streetcode.WebApi.Controllers.Source;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetCategoriesByStreetcodeIdTest.cs:23]: please remove unnecessary usings

File: Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetCategoriesByStreetcodeIdTest.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        private readonly Mock<IMapper> _mockMapper;
+        public GetCategoriesByStreetcodeIdTest()
+        {
+            _mockRepository = new Mock<IRepositoryWrapper>();
+            _mockMapper = new Mock<IMapper>();
+        }
+        [Fact]
+        public async Task ShouldReturnSuccesfully()
+        {
+
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetCategoriesByStreetcodeIdTest.cs:38]: try not to leave extra empty lines

File: Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetCategoriesByStreetcodeIdTest.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            _mockMapper.Setup(x => x.Map<IEnumerable<SourceLinkCategoryDTO>>(It.IsAny<IEnumerable<SourceLinkCategory>>()))
+                .Returns(GetSourceDTOs());
+
+            var handler = new GetCategoriesByStreetcodeIdHandler(_mockRepository.Object, _mockMapper.Object);
+
+            var expectedError = $"Cant find any source category with the streetcode id {id}";
+            // act
+
+            var result = await handler.Handle(new GetCategoriesByStreetcodeIdQuery(1), CancellationToken.None);
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetCategoriesByStreetcodeIdTest.cs:85]: it is a better practice to keep values in (test) variables - avoid using "magical" numbers

File: Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetCategoryByIdTest.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+        // act
+
+        var result = await handler.Handle(new GetCategoryByIdQuery(1), CancellationToken.None);
+
+        // assert
+        Assert.Multiple(
+        () => Assert.NotNull(result),
+        () => Assert.IsType<SourceLinkCategoryDTO>(result.ValueOrDefault)
+        );
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetCategoryByIdTest.cs:48]: maybe some extra 'tab' spacings on 46, 47 line will make code prittier

File: Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetSubCategoriesByCategoryIdTest.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+using Streetcode.DAL.Entities.Sources;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetSubCategoriesByCategoryIdTest.cs:17]: same here

File: Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetSubCategoriesByCategoryIdTest.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                .Returns(GetSourceSubDTOs());
+
+            // arrange
+
+            var handler = new GetSubCategoriesByCategoryIdHandler(_mockRepository.Object, _mockMapper.Object);
+
+            // act
+
+            var result = await handler.Handle(new GetSubCategoriesByCategoryIdQuery(1), CancellationToken.None);
+
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetSubCategoriesByCategoryIdTest.cs:49]: same here

File: Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetSubCategoriesByCategoryIdTest.cs [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            Assert.Multiple(
+            () => Assert.NotNull(result),
+            () => Assert.IsType<List<SourceLinkSubCategoryDTO>>(result.ValueOrDefault)
+            ) ;
+        }
+
+        [Theory]
+        [InlineData(1)]
+        public async Task ShouldReturnNull_NotExistingId(int id = 1)
+        {
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetSubCategoriesByCategoryIdTest.cs:59]: you can remove this defauilt declaration (you already have inline data instead)

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/WithIndexExistHandlerTest.cs [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using Streetcode.DAL.Repositories.Realizations.Streetcode;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Sdk;
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/WithIndexExistHandlerTest.cs:18]: same here

**Review by Katerix — CHANGES_REQUESTED** [review]
Good job for starters! Please read review comments and  change similar code in all necessary files)

**Review by Katerix — APPROVED** [review]
Awesome!

**Pull request merged**